src/main/java/com/redhat/devtools/intellij/tektoncd/listener/SaveInEditorListener.java

```diff
@@ -50,19 +55,16 @@
     @Override
     public boolean maySaveDocument(@NotNull Document document, boolean isSaveExplicit) {
         VirtualFile vf = FileDocumentManager.getInstance().getFile(document);
-
-        // if file is not related to tekton we can skip it
-        if (vf == null || vf.getUserData(KIND_PLURAL).isEmpty()) {
-            return true;
-        }
+        Project project = vf.getUserData(PROJECT);
+        if(project == null || !isFileToPush(project, document, vf)) return true;
 
         String namespace, name, apiVersion;
         JsonNode spec;
         CustomResourceDefinitionContext crdContext;
         Notification notification;
         try {
             namespace = YAMLHelper.getStringValueFromYAML(document.getText(), new String[] {"metadata", "namespace"});
-            if (Strings.isNullOrEmpty(namespace)) {
+            if ((namespace == null && !vf.getUserData(KIND_PLURAL).equals(KIND_CLUSTERTASKS)) || namespace == "") {
```

Comment:
Note sure to agree with this test as it will allow a namespace for a cluster task